src/pk/ecc/ecc_cmp_oid_str.c

```diff
@@ -0,0 +1,40 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt_private.h"
+
+#ifdef LTC_MECC
+
+/** Compare curve's OID to constant string
+  @param oidstr [in] the desired OID in string format
+  @param key    The ECC key
+  Return        CRYPT_OK if the curve has the correct OID
+*/
+
+int ecc_cmp_oid_str(const char *oidstr, const ecc_key *key)
+{
+   char oid[64];
+   unsigned long oidlen = sizeof(oid);
+   int err;
+
+   LTC_ARGCHK(key != NULL);
+
+   if ((err = pk_oid_num_to_str(key->dp.oid, key->dp.oidlen, oid, &oidlen)) != CRYPT_OK)                { goto error; }
```

Comment:
if we merge this after #404 there's going to be `int pk_oid_cmp_with_ulong(const char *o1, const unsigned long *o2, unsigned long o2size);` ;)

Reply:
I can drop `ecc_cmp_oid_str` entirely, then. Turned out that it was only needed for the secp256k1 checks for `LTC_ECCSIG_ETH27`. Let me know if we want to do that.

Reply:
We keep it for now as is I'd say as the other PR seems to take some more iterations as it's not entirely ready yet.

Reply:
Even better, simply cherry-pick those commits and whoever gets merged first will have them and the second will be rebased and the commits removed :)  Git FTW

Reply:
Took a while to figure out the syntax - thought I'd have to name the branch I was pulling from - but that was remarkably painless! 🍒 

I'll get rid of `ecc_cmp_oid_str` now.